src/algo/depth-to-rgb-calibration/calibration.cpp

```diff
@@ -0,0 +1,387 @@
+
+
+calib::calib( rs2_intrinsics_double const & intrin, rs2_extrinsics_double const & extrin )
+{
+    auto const & r = extrin.rotation;
+    auto const & t = extrin.translation;
+    auto const & c = intrin.coeffs;
+
+    height = intrin.height;
+    width = intrin.width;
+    rot = { r[0], r[1], r[2], r[3], r[4], r[5], r[6], r[7], r[8] };
+    trans = { t[0], t[1], t[2] };
+    k_mat = matrix_3x3{ intrin.fx, 0,intrin.ppx,
+        0,intrin.fy, intrin.ppy,
+        0,0,1 };
+
+    coeffs[0] = c[0];
+    coeffs[1] = c[1];
+    coeffs[2] = c[2];
+    coeffs[3] = c[3];
+    coeffs[4] = c[4];
+    model = intrin.model;
+}
+
+calib::calib( rs2_intrinsics const & intrin, rs2_extrinsics const & extrin )
+{
+    auto const & r = extrin.rotation;
+    auto const & t = extrin.translation;
+    auto const & c = intrin.coeffs;
+
+    height = intrin.height;
+    width = intrin.width;
+    rot = { r[0], r[1], r[2], r[3], r[4], r[5], r[6], r[7], r[8] };
+    trans = { t[0], t[1], t[2] };
+    k_mat = matrix_3x3{ intrin.fx, 0,intrin.ppx,
+        0,intrin.fy, intrin.ppy,
+        0,0,1 };
+    coeffs[0] = c[0];
+    coeffs[1] = c[1];
+    coeffs[2] = c[2];
+    coeffs[3] = c[3];
+    coeffs[4] = c[4];
+    model = intrin.model;
+}
+
+rs2_intrinsics_double calib::get_intrinsics() const
+{
+    return {
+        width, height,
+        k_mat,
+        model, coeffs };
+}
+
+rs2_extrinsics_double calib::get_extrinsics() const
+{
+    auto & r = rot.rot;
+    auto & t = trans;
+    return {
+        { r[0] , r[1], r[2], r[3], r[4], r[5], r[6], r[7], r[8] },
+        { t.t1, t.t2, t.t3 }
+    };
+}
+
+p_matrix const calib::calc_p_mat() const
+{
+    auto r = rot.rot;
+    auto t = trans;
+    auto k = k_mat.k_mat.rot;
+   /* auto fx = k_mat.fx;
+    auto fy = k_mat.fy;
+    auto ppx = k_mat.ppx;
+    auto ppy = k_mat.ppy;*/
+    p_matrix p_mat = {
+        k[0] * r[0] + k[1] * r[3] + k[2] * r[6], 
+        k[0] * r[1] + k[1] * r[4] + k[2] * r[7], 
+        k[0] * r[2] + k[1] * r[5] + k[2] * r[8], 
+        k[0] * t.t1 + k[1] * t.t2 + k[2] * t.t3,
+
+        k[3] * r[0] + k[4] * r[3] + k[5] * r[6], 
+        k[3] * r[1] + k[4] * r[4] + k[5] * r[7], 
+        k[3] * r[2] + k[4] * r[5] + k[5] * r[8], 
+        k[3] * t.t1 + k[4] * t.t2 + k[5] * t.t3,
+
+        r[6], 
+        r[7], 
+        r[8], 
+        t.t3 
+    };
+
+    return p_mat;
+}
+
+void calib::copy_coefs( calib & obj ) const
+{
+    obj.width = this->width;
+    obj.height = this->height;
+
+    obj.coeffs[0] = this->coeffs[0];
+    obj.coeffs[1] = this->coeffs[1];
+    obj.coeffs[2] = this->coeffs[2];
+    obj.coeffs[3] = this->coeffs[3];
+    obj.coeffs[4] = this->coeffs[4];
+
+    obj.model = this->model;
+}
+
+calib calib::operator*( double step_size ) const
+{
+    calib res;
+    for (auto i = 0; i < 9; i++)
+    {
+        res.k_mat.k_mat.rot[i] = k_mat.k_mat.rot[i] * step_size;;
+    }
+
+    for (auto i = 0; i < 9; i++)
+        res.rot.rot[i] = this->rot.rot[i] * step_size;
+
```

Comment:
The code repeats itself in all +-/* operations:
```cpp
    calib res;
    for (auto i = 0; i < 9; i++)
    {
        res.k_mat.k_mat.rot[i] = k_mat.k_mat.rot[i] * step_size;;
    }

    for (auto i = 0; i < 9; i++)
        res.rot.rot[i] = this->rot.rot[i] * step_size;
```
Can this be refactored into  ```calib res{k_mat , this->rot };``` or similar?

Reply:
I see the loop repeating, but with a different operation... maybe I'm misunderstanding?

Reply:
For all those operators it always starts with the about segment (C&P) that I highlighted. Only the last 3-6 lines of code differ.
so those specific lines could be refactored